cloud/scope/powervs_cluster_test.go

```diff
@@ -3043,3 +3045,958 @@ func TestCreateVPCSubnet(t *testing.T) {
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					TransitGateway: &infrav1beta2.TransitGatewayStatus{
+						ID: ptr.To("transitGatewayID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGateway(gomock.Any()).Return(&tgapiv1.TransitGateway{Name: ptr.To("transitGatewayName"), Status: ptr.To("failed")}, nil, nil)
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When TransitGatewayID is set in Status, returns Transit Gateway and is in pending state", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					TransitGateway: &infrav1beta2.TransitGatewayStatus{
+						ID: ptr.To("transitGatewayID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGateway(gomock.Any()).Return(&tgapiv1.TransitGateway{Name: ptr.To("transitGatewayName"), Status: ptr.To("pending")}, nil, nil)
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeTrue())
+		g.Expect(err).To(BeNil())
+	})
+
+	t.Run("When TransitGatewayID is set in Spec, returns error while getting transit Gateway", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					TransitGateway: &infrav1beta2.TransitGateway{
+						ID: ptr.To("transitGatewayID"),
+					},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGateway(gomock.Any()).Return(nil, nil, errors.New("failed to get transit gateway"))
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When TransitGatewayID is not set, transit Gateway is in failed state and return error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					TransitGateway: &infrav1beta2.TransitGateway{},
+				},
+			},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGatewayByName(gomock.Any()).Return(&tgapiv1.TransitGateway{Name: ptr.To("transitGatewayName"), ID: ptr.To("transitGatewayID"), Status: ptr.To("failed")}, nil)
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When TransitGatewayID is not set, returns transit Gateway and is in pending state", func(t *testing.T) {
```

Comment:
same here as well, should we check both failed and pending state?

Reply:
Here these are covering this part of code 
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/cloud/scope/powervs_cluster.go#L1670
- When TransitGateway is in failed state, it will return error
- when it is in pending state it will just return true